libs/ui/forms/src/lib/upload.tsx

```diff
@@ -145,17 +197,43 @@ function tryGetMediaType(file: File): string | null {
 	return null;
 }
 
-async function uploadFile(file: File, url: string): Promise<void> {
-	const res = await fetch(url, {
-		method: "PUT",
-		body: file
-	});
-
-	console.log("Upload response:", res);
+async function uploadWithProgress(
+	url: string,
+	file: File,
+	showDialog?: () => void,
+	onProgress?: (bytes: number) => void,
+	onComplete?: (e: ProgressEvent<XMLHttpRequestEventTarget>) => void
```

Comment:
optional parameter wirklich notwendig? 

wenn sie nicht gesetzt werden macht die methode nichts. und in dem einen aufruf werden sie gesetzt. 

ich vermute dass man die bei vernünftigen UI design vermutlich auch immer haben möchte?

Reply:
Die Methode ist so gestaltet, dass wir sie prinzipiell auch anderswo verwenden könnten (daher mit export attribut versehen). Allerdings gibt es aktuell keinen anderen Anwendungszweck und könnte daher angepasst werden.

Prinzipiell sind die 3 optionalen Parameter durchauch optional, da es hier um das setzen von Listenern geht (Progressbar & Progressbar Sichtbarschaltung. Wenn die Parameter nicht gesetzt sind, funktioniert Upload wie vorher auch, ohne Rückmeldung.